invoices/invoiceregistry.go

```diff
@@ -856,63 +814,85 @@ func (i *InvoiceRegistry) NotifyExitHopHtlc(rHash lntypes.Hash,
 		return nil, err
 	}
 
-	debugLog(result.String())
-
 	if updateSubscribers {
 		i.notifyClients(rHash, invoice, invoice.State)
 	}
 
-	// Inspect latest htlc state on the invoice.
-	invoiceHtlc, ok := invoice.Htlcs[circuitKey]
+	switch res := resolution.(type) {
+	case *HtlcFailResolution:
+		// Inspect latest htlc state on the invoice. If it is found,
+		// we will update the accept height as it was recorded in the
+		// invoice database (which occurs in the case where the htlc
+		// reached the database in a previous call). If the htlc was
```

Comment:
previously `currentHeight` was used if it was not found. Do we still want to do that?

Reply:
Yes, I think it should be the current height indeed. It is also reported in the `invalid_details` failure message.

Reply:
`currentHeight` is set when the resolution is created (in `updateInvoice`), so it will still be set after this change. Future refactor will aim to move everything in there, so we don't need to set any values on the resolution like we do here.